docs/architecture/third-party/drupal/testing.md

```diff
@@ -0,0 +1,87 @@
+# Testing Drupal components
+
+OpenEuropa requires automated tests to be written for every new feature or
+bugfix to ensure that the functionality will keep working as expected in the
+future.
+
+* Pull requests without sufficient test coverage will not be accepted.
```

Comment:
Do we have a metric on what sufficient test coverage means?

Reply:
There is only one metric that I know of and that's to do code coverage analysis. Unfortunately that's a flawed metric that doesn't say anything about the quality of the tests, and in my opinion unreasonable to ask. For example requiring 100% coverage would force people to write tests for trivial code like getters and setters which is time wasted that could be spent on delivering actual value.

I think this is hard to define, and probably at the discretion of the reviewers.

Reply:
A bit philosophical here, but I was wondering what is the good approach in our case? 
- a kind of 80/20 functionality trivial/special cases: 80% trivial cases covered by 20% of scenarios and 20% special cases covered by 80% of scenarios.
- or minimum viable : just the 3 or max 5 scenarios that covers the core business functionality to deliver and for the special cases (not yet identified by business or dev team) it will come with the usage an handled as bug fixes....

Reply:
Trivial code means things that have 0 business value, like class constructors, wrappers, factory methods, getters and setters.

I firmly believe that if there is business logic then it should always be tested, 100% of it. If it is not tested it cannot be deployed in good confidence.